client/packages/common/src/ui/forms/JsonForms/components/CategorizationTabLayout.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { FC, useState } from 'react';
+import {
+  and,
+  Categorization,
+  categorizationHasCategory,
+  Category,
+  isVisible,
+  LayoutProps,
+  optionIs,
+  RankedTester,
+  rankWith,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { withJsonFormsLayoutProps } from '@jsonforms/react';
+import { Button, Grid } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { DialogButton } from '../../../components/buttons';
+import { ModalProps, useDialog } from '@common/hooks';
+import {
+  AjvProps,
+  MaterialLayoutRenderer,
+  MaterialLayoutRendererProps,
+  withAjvProps,
+} from '@jsonforms/material-renderers';
+
+interface CategoryModalProps extends ModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const CategoryModal: FC<CategoryModalProps> = ({
+  children,
+  isOpen,
+  onClose,
+  ...modalProps
+}) => {
+  const { Modal } = useDialog({
+    isOpen,
+    onClose,
+  });
+  return <Modal {...modalProps}>{children}</Modal>;
+};
+
+export const categorizationTabLayoutTester: RankedTester = rankWith(
+  2,
+  and(
+    uiTypeIs('Categorization'),
+    categorizationHasCategory,
+    optionIs('variant', 'tab')
+  )
+);
+
+const Icon = styled('i')(({ theme }) => ({
+  backgroundColor: theme.palette.primary.main,
+  height: 50,
+  width: 50,
+}));
+
+const UIComponent: FC<LayoutProps & AjvProps> = ({
+  ajv,
+  data,
+  path,
+  renderers,
+  schema,
+  uischema,
+  visible,
+  cells,
+}) => {
+  const [activeCategory, setActiveCategory] = useState<number | undefined>();
+  const categorization = uischema as Categorization;
+
+  const categories = categorization.elements.filter(
+    (category: Category | Categorization): category is Category =>
+      isVisible(category, data, '', ajv) && category.type === 'Category'
+  );
```

Comment:
I made a change using a type guard instead of the extra map here...

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates